index.js

```diff
@@ -0,0 +1,116 @@
+// INSTALL DEPENDENCIES 
+const express = require('express'); // REQUIRE EXPRESS
+const server = express(); // CREATE SERVER
+const helmet = require('helmet'); // REQUIRE HELMET
+const knex = require('knex') // REQUIRE KNEX 
+
+// DEFINE KNEX CONFIGURATION
+const knexConfig = {
+    client: 'sqlite3',
+    connection: {
+        filename: './data/projectx.db3',
+    },
+    useNullAsDefault: true, // needed for sqlite
+    };
+    
+// DEFINE DATABASE CONFIGURATION
+const dbConfig = require("./knexfile")
+// DEFINE DATABASE
+const db = knex(dbConfig.development);
+
+
+// TEST ROUTE HANDLER FUNCTION
+server.get('/', (req, res) => {
+  res.send('Server is running ');
+});
+
+// SET PORT 
+const port = process.env.PORT || 9000; 
+
+// USE MIDDLEWARE
+server.use(express.json())
+
+// TEST
+server.get('/api/test', (req, res) => {
+    db.select().from('projects').then(data => res.json(data)).catch(err => console.log(err))
+});
+
+// RETRIEVE ALL PROJECTS
+server.get('/api/projects', (req, res, next) => {
+    db('projects')
+        .then(data => res.status(200).json(data))
+        .catch(next)
+});
+
+// RETREIVE PROJECT BY ID
+server.get('/api/projects/:id', (req, res, next) => {
+const { id } = req.params;
+    db.select()
+        .from('projects')
+        .where({ id })
+        .then(projectArr => {
+            db.select()
+                .from('actions')
+                .where('project_id', id)
+                .then(actions => {
+                const project = projectArr[0];
+            res.status(200).json({
+                id: project.id,
+                name: project.name,
+                description: project.description,
+                completed: project.completed ? true : false,
+                actions: actions.map(action => {
+                return {
+                    id: action.id,
+                    description: action.description,
+                    notes: action.notes,
```

Comment:
I don't think you need to map over the array of actions here. You get back an array of actions by project id and then you just append it to the project object